src/Npgsql/NpgsqlConnection.cs

```diff
@@ -510,6 +510,12 @@ public new NpgsqlTransaction BeginTransaction(IsolationLevel level)
             var connector = CheckReadyAndGetConnector();
             Debug.Assert(Connector != null);
 
+            // If connected to a backend that does not support transactions, ignore the
+            // call to BeginTransaction silently. Some layers (e.g. Entity Framework) may
+            // automatically start transactions and we want to avoid errors for that.
+            if (!connector.SupportsTransactions)
```

Comment:
This will not work (I think at some point I already pointed this out). An application calling `BeginTransaction()` expets to receive a DbTransaction instance back, on which is calls `Commit()` or `Rollback()` - that will throw a NullReferenceException.

To silently ignore transactions, you need `BeginTransaction()` to return a valid instance of NpgsqlTransaction, but make sure nothing is actually communicated to the backend at any point.